internal/api/migrate/loader.go

```diff
@@ -65,13 +65,15 @@ func (c *migCuePart) GetSchemaPath() string {
 	return c.schemasPath
 }
 
-func (c *migCuePart) Load() error {
-	conditions, err := c.buildListOfConditions()
+// NB There are no relevant case of "failed loads" in the case of migration schemas at the moment
+func (c *migCuePart) Load() (successfulLoadsCount, _ int, err error) {
+	var conditions string
+	conditions, successfulLoadsCount, err = c.buildListOfConditions()
 	if err != nil {
-		return err
+		return
 	}
 	c.listOfConditions.Store(&conditions)
-	return nil
+	return
```

Comment:
More readable if you keep not named returns no ? 

```go
	conditions, successfulLoadsCount, err := c.buildListOfConditions()
	if err == nil {
    c.listOfConditions.Store(&conditions)
	}
  return successfulLoadsCount, 0, err
```

Reply:
btw if I understand well the second returns is the failure counts? 
Migration can't fail?

Reply:
> More readable if you keep not named returns no ?
> 
> ```go
> 	conditions, successfulLoadsCount, err := c.buildListOfConditions()
> 	if err == nil {
>           c.listOfConditions.Store(&conditions)
> 	}
>         return successfulLoadsCount, 0, err
> ```

I was hesitating to do this, invert the if condition so that we only have 1 return clause, but it may be confusing to have a `err == nil` check while 99% of the time the usual pattern is `err != nil`. I had the feeling that could be the kind of thing one could miss at a first glance. But maybe I'm worrying too much, WDYT?

Reply:
> btw if I understand well the second returns is the failure counts? Migration can't fail?

Yes kinda, that's what I put in comment above (and in PR desc too)
> // NB There are no relevant case of "failed loads" in the case of migration schemas at the moment

Basically regarding migration schema, since they are not mandatory we just skip if we don't find one. In the current state of the code there's no relevant error case to monitor. I suppose that we could some validation based on the format we expect (explained in the doc https://github.com/perses/perses/blob/main/docs/plugin/cue.md#variable-1), at least to validate that we indeed have a list of conditions for instance.
I wont tackle this for now but I'll add a comment to mention the future work at least.

Reply:
> I was hesitating to do this, invert the if condition so that we only have 1 return clause, but it may be confusing to have a err == nil check while 99% of the time the usual pattern is err != nil. I had the feeling that could be the kind of thing one could miss at a first glance. But maybe I'm worrying too much, WDYT?

I still find more readable my way but I understand it might be subjective. As you wish!